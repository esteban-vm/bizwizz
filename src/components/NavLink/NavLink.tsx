import type { FC, UUID, SectionID } from '@/types'
import Link from 'next/link'

export interface NavLinkProps {
  id: UUID
  text: Capitalize<string>
  to: `#${Extract<SectionID, 'contact' | 'portfolio' | 'services'>}`
}

const NavLink: FC<NavLinkProps> = ({ to, text, ...props }) => {
  return (
    <li className='nav_link' {...props}>
      <Link href={to}>{text}</Link>
    </li>
  )
}

export default NavLink
