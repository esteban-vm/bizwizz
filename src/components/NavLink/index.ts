export { default as NavLink, type NavLinkProps } from './NavLink'
