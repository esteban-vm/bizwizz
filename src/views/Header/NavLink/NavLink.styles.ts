import clsx from 'clsx'

const styles = clsx(
  'ml-10 block select-none pb-[0.8rem] ',
  'text-[1.4rem] uppercase text-white transition-[border-b]',
  'hover:border-b-2 hover:border-b-primary'
)
export default styles
