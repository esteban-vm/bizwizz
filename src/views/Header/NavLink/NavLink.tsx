import type { FC, UUID } from '@/types'
import Link from 'next/link'
import styles from './NavLink.styles'

export type NavLinkProps = {
  id: UUID
  link: `#${Lowercase<string>}`
  text: Capitalize<string>
}

const NavLink: FC<NavLinkProps> = ({ id, link, text }) => {
  return (
    <li id={id}>
      <Link className={styles} href={link}>
        {text}
      </Link>
    </li>
  )
}

export default NavLink
