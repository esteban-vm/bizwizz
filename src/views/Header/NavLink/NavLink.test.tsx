import { render, cleanup, within, screen } from '@/tests'
import NavLink from './NavLink'
import navLinks from '../Header.data'

describe('Nav Link:', () => {
  let asFragment: () => DocumentFragment
  const [testNavLink] = navLinks

  beforeEach(() => {
    void ({ asFragment } = render(<NavLink {...testNavLink} />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const navLinkElem = screen.getByRole('listitem')
    expect(navLinkElem).toBeInTheDocument()
    expect(navLinkElem).toHaveAttribute('id', testNavLink.id)
    const linkElem = within(navLinkElem).getByRole('link', { name: testNavLink.text })
    expect(linkElem).toBeInTheDocument()
    expect(linkElem).toHaveAttribute('href', testNavLink.link)
  })
})
