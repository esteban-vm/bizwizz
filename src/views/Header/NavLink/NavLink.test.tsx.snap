// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav Link: > should render correctly 1`] = `
<DocumentFragment>
  <li
    id="c054a01e-9103-4cd7-8a2f-c647d0d3e73b"
  >
    <a
      class="
          ml-10
          block
          select-none
          pb-[0.8rem]
          text-[1.4rem]
          uppercase
          text-white
          transition-[border-b]
          hover:border-b-2
          hover:border-b-primary
        "
      href="#home"
    >
      Home
    </a>
  </li>
</DocumentFragment>
`;
